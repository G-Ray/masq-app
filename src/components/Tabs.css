.icon path {
  fill: #707070;
}

.tab {
  display: flex;
  align-items: center;
  cursor: default;
}

.active {
  color: #6ba3fa;
}

.active path {
  fill: #6ba3fa;
}
