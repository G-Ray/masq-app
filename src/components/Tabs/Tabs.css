.Tabs {
  height: 65px;
  display: flex;
  align-items: center;
  position: relative;
  justify-content: space-evenly;
  background-color: #f5f7fa;
  color: #707070;
}

.Tabs #tab {
  display: flex;
  align-items: center;
  cursor: default;
}

.Tabs p {
  margin-left: 8px;
}

.Tabs .active {
  color: #6ba3fa;
}
