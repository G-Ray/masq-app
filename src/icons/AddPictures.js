import React from 'react'

const AddPictures = props => (
  <svg height={120} viewBox='0 0 120 120' width={120} {...props}>
    <g fill='none' fillRule='evenodd'>
      <circle cx={60} cy={60} fill='#fff' r={60} />
      <circle cx={60} cy={60} r={59} stroke='currentColor' strokeWidth={2} />
      <path
        d='M76.8 46.933h-6.533l-3.174-4.853c-.373-.373-.933-.747-1.493-.747H54.4c-.56 0-1.12.374-1.493.747l-3.174 4.853H43.2c-3.173 0-5.6 2.427-5.6 5.6v20.534c0 3.173 2.427 5.6 5.6 5.6h33.6c3.173 0 5.6-2.427 5.6-5.6V52.533c0-3.173-2.427-5.6-5.6-5.6zm1.867 26.134c0 1.12-.747 1.866-1.867 1.866H43.2c-1.12 0-1.867-.746-1.867-1.866V52.533c0-1.12.747-1.866 1.867-1.866h7.467c.56 0 1.12-.374 1.493-.747l3.173-4.853h9.147l3.173 4.853c.56.373 1.12.747 1.68.747H76.8c1.12 0 1.867.746 1.867 1.866zM60 52.533c-5.227 0-9.333 4.107-9.333 9.334 0 5.226 4.106 9.333 9.333 9.333s9.333-4.107 9.333-9.333c0-5.227-4.106-9.334-9.333-9.334zm0 14.934c-3.173 0-5.6-2.427-5.6-5.6 0-3.174 2.427-5.6 5.6-5.6s5.6 2.426 5.6 5.6c0 3.173-2.427 5.6-5.6 5.6z'
        fill='currentColor'
        fillRule='nonzero'
      />
    </g>
  </svg>
)

export default AddPictures
