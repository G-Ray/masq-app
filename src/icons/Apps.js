import React from 'react'

const Apps = props => (
  <svg
    xmlnsXlink='http://www.w3.org/1999/xlink'
    width={20}
    height={20}
    viewBox='0 0 20 20'
    {...props}
  >
    <defs>
      <path
        id='apps-svg'
        d='M17 0H3C1.3 0 0 1.3 0 3v14c0 1.7 1.3 3 3 3h14c1.7 0 3-1.3 3-3V3c0-1.7-1.3-3-3-3zM3 2h14c.6 0 1 .4 1 1v3H2V3c0-.6.4-1 1-1zM2 17V8h4v10H3c-.6 0-1-.4-1-1zm15 1H8V8h10v9c0 .6-.4 1-1 1z'
      />
    </defs>
    <use fill='currentColor' fillRule='nonzero' xlinkHref='#apps-svg' />
  </svg>
)

export default Apps
