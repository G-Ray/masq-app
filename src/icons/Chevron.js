import React from 'react'

const Chevron = props => (
  <svg
    xmlnsXlink='http://www.w3.org/1999/xlink'
    width={14}
    height={10}
    viewBox='0 0 14 10'
    {...props}
  >
    <defs>
      <path
        id='chevron-svg'
        d='M13.7 1.7l-6 6c-.2.2-.4.3-.7.3-.3 0-.5-.1-.7-.3l-6-6C-.1 1.3-.1.7.3.3c.4-.4 1-.4 1.4 0L7 5.6 12.3.3c.4-.4 1-.4 1.4 0 .4.4.4 1 0 1.4z'
      />
    </defs>
    <use fill='currentColor' fillRule='nonzero' xlinkHref='#chevron-svg' />
  </svg>
)

export default Chevron
