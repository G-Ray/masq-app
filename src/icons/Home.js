import React from 'react'

const Home = props => (
  <svg
    xmlnsXlink='http://www.w3.org/1999/xlink'
    width={20}
    height={22}
    viewBox='0 0 20 22'
    {...props}
  >
    <defs>
      <path
        id='home-svg'
        d='M19.6 7.217l-9-6.992c-.4-.3-.9-.3-1.2 0l-9 6.992c-.3.2-.4.5-.4.799v10.987C0 20.701 1.3 22 3 22h14c1.7 0 3-1.299 3-2.997V8.016c0-.3-.1-.6-.4-.8zM12 20.002H8v-7.99h4v7.99zm6-.999c0 .6-.4 1-1 1h-3v-8.99c0-.6-.4-1-1-1H7c-.6 0-1 .4-1 1v8.99H3c-.6 0-1-.4-1-1V8.515l8-6.193 8 6.193v10.488z'
      />
    </defs>
    <use fill='currentColor' fillRule='nonzero' xlinkHref='#home-svg' />
  </svg>
)

export default Home
