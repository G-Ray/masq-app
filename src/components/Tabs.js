import React from 'react';
import ReactSVG from 'react-svg';
import { withRouter } from 'react-router-dom';

import './Tabs.css';

const style = {
  height: '65px',
  backgroundColor: '#f5f7fa',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-evenly',
  color: '#707070',
  position: 'relative'
  // boxShadow: '0 0 8px 0 rgba(0, 0, 0, 0.15)'
};

class Tabs extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tabs: [],
      activeTab: 0
    }
  }

  onSelectTab(key) {
    console.log('onSelectTab')
    const { history, tabs } = this.props;

    this.setState({ activeTab: key });
    history.push(tabs[key].link);
  }

  render() {
    const { activeTab } = this.state;

    return (
      <div style={style}>
        {this.props.tabs.map((tab, index) => (
          <div className={ 'tab ' + (index === activeTab ? 'active' : 'inactive') }
               key={index}
               onClick={() => this.onSelectTab(index)}
          >
            <ReactSVG className="icon" path={tab.icon} />
            {/* <img src={tab.icon} alt={tab.icon} /> */}
            <p style={{marginLeft: '8px'}}>{tab.label}</p>
          </div>
        ))}
      </div>
    );
  }
}

export default withRouter(Tabs)
