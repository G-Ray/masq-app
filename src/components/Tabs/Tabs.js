import React from 'react'
import { withRouter } from 'react-router-dom'

import './Tabs.css'

const style = {
  height: '65px',
  backgroundColor: '#f5f7fa',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-evenly',
  color: '#707070',
  position: 'relative'
  // boxShadow: '0 0 8px 0 rgba(0, 0, 0, 0.15)'
}

class Tabs extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      tabs: [],
      activeTab: 0
    }
  }

  onSelectTab (key) {
    const { history, tabs } = this.props

    this.setState({ activeTab: key })
    history.push(tabs[key].link)
  }

  render () {
    const { activeTab } = this.state

    return (
      <div style={style}>
        {this.props.tabs.map((tab, index) => (
          <div className={'tab ' + (index === activeTab ? 'active' : 'inactive')}
            key={index}
            onClick={() => this.onSelectTab(index)}
          >
            { tab.icon }
            <p style={{marginLeft: '8px'}}>{tab.label}</p>
          </div>
        ))}
      </div>
    )
  }
}

export default withRouter(Tabs)
