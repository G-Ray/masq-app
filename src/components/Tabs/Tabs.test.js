import React from 'react'
import renderer from 'react-test-renderer'
import Tabs from './Tabs'
import { MemoryRouter } from 'react-router'

import { Smartphone, Apps } from 'icons'

const tabs = [
  { label: 'Devices', link: '/devices', icon: <Smartphone /> },
  { label: 'Applications', link: '/applications', icon: <Apps /> }
]

it('renders Tabs with one tabs', () => {
  const component = renderer.create(
    <MemoryRouter><Tabs tabs={[tabs[0]]} /></MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Tabs with two tabs', () => {
  const component = renderer.create(
    <MemoryRouter><Tabs tabs={tabs} /></MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
