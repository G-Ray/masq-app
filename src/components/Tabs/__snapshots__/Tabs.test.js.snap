// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tabs with one tabs 1`] = `
<div
  className="Tabs"
>
  <div
    className="active"
    id="tab"
    onClick={[Function]}
  >
    <svg
      height={22}
      viewBox="0 0 16 22"
      width={16}
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <path
          d="M13 0H3C1.3 0 0 1.3 0 3v16c0 1.7 1.3 3 3 3h10c1.7 0 3-1.3 3-3V3c0-1.7-1.3-3-3-3zm1 19c0 .6-.4 1-1 1H3c-.6 0-1-.4-1-1V3c0-.6.4-1 1-1h10c.6 0 1 .4 1 1v16zm-6-1a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
          id="smartphone-svg"
        />
      </defs>
      <use
        fill="currentColor"
        fillRule="nonzero"
        xlinkHref="#smartphone-svg"
      />
    </svg>
    <p>
      DEVICES
    </p>
  </div>
</div>
`;

exports[`renders Tabs with two tabs 1`] = `
<div
  className="Tabs"
>
  <div
    className="active"
    id="tab"
    onClick={[Function]}
  >
    <svg
      height={22}
      viewBox="0 0 16 22"
      width={16}
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <path
          d="M13 0H3C1.3 0 0 1.3 0 3v16c0 1.7 1.3 3 3 3h10c1.7 0 3-1.3 3-3V3c0-1.7-1.3-3-3-3zm1 19c0 .6-.4 1-1 1H3c-.6 0-1-.4-1-1V3c0-.6.4-1 1-1h10c.6 0 1 .4 1 1v16zm-6-1a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
          id="smartphone-svg"
        />
      </defs>
      <use
        fill="currentColor"
        fillRule="nonzero"
        xlinkHref="#smartphone-svg"
      />
    </svg>
    <p>
      DEVICES
    </p>
  </div>
  <div
    className="inactive"
    id="tab"
    onClick={[Function]}
  >
    <svg
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <path
          d="M17 0H3C1.3 0 0 1.3 0 3v14c0 1.7 1.3 3 3 3h14c1.7 0 3-1.3 3-3V3c0-1.7-1.3-3-3-3zM3 2h14c.6 0 1 .4 1 1v3H2V3c0-.6.4-1 1-1zM2 17V8h4v10H3c-.6 0-1-.4-1-1zm15 1H8V8h10v9c0 .6-.4 1-1 1z"
          id="apps-svg"
        />
      </defs>
      <use
        fill="currentColor"
        fillRule="nonzero"
        xlinkHref="#apps-svg"
      />
    </svg>
    <p>
      APPLICATIONS
    </p>
  </div>
</div>
`;
